src/SFML/Graphics/View.cpp

```diff
@@ -87,6 +87,18 @@ void View::setViewport(const FloatRect& viewport)
 }
 
 
+////////////////////////////////////////////////////////////
+void View::setScissor(const FloatRect& scissor)
+{
+    assert(scissor.left >= 0.0f && scissor.left <= 1.0f && "scissor.left must lie within [0, 1]");
+    assert(scissor.top >= 0.0f && scissor.top <= 1.0f && "scissor.top must lie within [0, 1]");
+    assert(scissor.width >= 0.0f && scissor.width <= 1.0f && "scissor.width must lie within [0, 1]");
+    assert(scissor.height >= 0.0f && scissor.height <= 1.0f && "scissor.height must lie within [0, 1]");
```

Comment:
Maybe this should be checking position+size coordinates instead of size.

Reply:
```cpp
    assert(sf::FloatRect({0, 0}, {1, 1}).contains(scissor.getPosition()) &&
           "View::setScissor() Scissor start position must lie within the screen");
    assert(sf::FloatRect({0, 0}, {1, 1}).contains(scissor.getPosition() + scissor.getSize()) &&
           "View::setScissor() Scissor end position must lie within the screen");
```
This is what I came up with for ensuring both the start and end of this `FloatRect` are within the screen.

Reply:
That would not work because `Rect::contains` checks if point is in the product of half-open intervals [left, left+width)*[top, top+height). In particular `sf::FloatRect({0, 0}, {1, 1}).contains({1, 1})` is false.
However it could be nicely expressed if we had a contains method between two Rects.

Reply:
This might be a place where having some Rect::getAbsolute method in the future could be useful to support negative size.